src/lib/Preview.js

```diff
@@ -1423,14 +1423,18 @@ class Preview extends EventEmitter {
             .then(() => {
                 // Reset retry count after successfully logging
                 this.logRetryCount = 0;
-                this.viewer.emit('preview_event_report', true);
+                if (this.viewer) {
```

Comment:
I'm not super familiar with the issue at hand, but I'm a bit confused by this change. If `this.viewer` is falsy, then `this.viewer.emit` couldn't be called in the first place and would throw rather than causing a loop. Have we actually prevented the issue here?

Reply:
due to reference error when this.viewer is undefined, it was falling into catch block where we have recurrent call of logPreviewEvent and on next call of logPreviewEvent again reference error .... with this if check we should be safe

Reply:
good question, there some small delay between document is loaded and /2.0/events request is finished so preview is happening for about 400ms
and by that do you mean to terminate pending an axios request or abandon the execution in _then_ block if this.viewer is undefined ?

Reply:
@arturfrombox disregard that last comment. I see what it's doing now.